### ISSUE TITLE
Alternative DDL Providers

### ISSUE THREAD

**Issue by rutskiuk** [mylar3/mylar3#546]
Would it be possible to add other DDL Providers, either by predefining them or allowing users to add their own?

I found some comics weren't getting results, particularly Manga so I searched elsewhere and found everything I needed so I thought it would be handy if this facility could be added.

Currently using:
 - W10 19041
 - Master 094147532c35c0704a83e7c6101c33f66c2335dc 

Examples of DDL providers: 
http://libgen.li/comics/
https://the-eye.eu/public/Comics/

**Comment by evilhero** [comment]
Probably not right off the hop. Adding in another DDL provider means a parser has to be built for each specific site, and then maintained going forward (cause one change in the html could lead to nothing working, which could happen quite frequently depending on the site).

libgen was being looked at by @barbequesauce at one point, but for the most part the time just isn't there for us to be adding in new providers that require a bit of work to get it going. That and honestly your results are going to be mixed if you're trying to get Manga with Mylar - the file parser is built to work with a normal-type of naming convention, and small subsets of changes therein. Manga is an entirely different beast as to how things are named, so the searcher/parser probably aren't going to get the results that you're going for (at least with Manga-related items), especially with CV being used as the main data source - and that's probably not going to change anytime soon.

**Comment by barbequesauce** [comment]
Yeah libgen was a no-go - while there are a number of parsers out there, none are current; there are a number of proxies as well, which all parse differently on top of that. It's a nonstarter.

**Comment by barbequesauce** [comment]
Closing due to staleness, we welcome contributions for new features. If this is an issue and you are still experiencing it, please open a new ticket with a care package attached.
